docs/appeng/adrs/0006-web-development-framework-preference.md

```diff
@@ -0,0 +1,92 @@
+
+In making a choice we're seeking to match up the tradeoffs that the tools provide with projected partnership needs. The technical ecosystems we operate in tend to change often, so this is a decision we should seek to re-evaluate at least annually. Some topics to consider in the decision include:
+
+- Accessibility for junior engineers.
+- Support for rapid prototyping.
+- Ability to use in 'enterprise' environments.
+- Batteries included vs. things we have to roll ourselves.
+  - e.g authentication and authorization tooling, data model management, caching, etc
+
+(this list is likely not complete, feel free to modify or add to it as we put this draft together)
+
+The decision outcome does not need to be a single framework. Ideally we'll choose 2-3 that we would ask folks to choose from in most scenarios.
+
+## Decision
+
+Put a decision outcome here when ready.
+
+## Why is this applicable to the Practice as a whole?
+
+A large portion of our work is focused on developing web apps in partnership with public and private organizations. Each time something new starts we need to make decisions about how we'd like to approach it. This is meant to help give guidance in order to lower the burden of that choice.
+
+Having a 'practice opinion' on these tools will also help support folks' professional development, as well as our hiring and on-boarding processes. As we hire and get folks up to speed, this decision may be able to help us tailor what we do toward 'what we're trying to be good at.' Similarly we could lean on this as we seek to provide resources for folks developing their professional development plans.
+
+## Alternatives Considered
+
+### Django
+
+#### Pros
+
+- `+` Mature framework that's been around since 2005.
+- `+` Opinionated tooling for managing data migrations, potentially making the task more accessible for junior practitioners.
+- `+` Strong ORM allowing the abstraction of many database interactions.
+- `+` Features designed to make form creation and management easier.
+- `+` Strong templating tooling
+- `+` Possible to gain a portion of the features typically associated with single page applications, with less overhead by pairing the framework with something like [htmx](https://htmx.org/).
+- `+` Uses the python programming language which enjoys broad community support, and is useful across multiple subpractices like web development and data science.
+- `+` Provides batteries included tooling for authentication and authorization, caching, and an administrative console, among others.
+- `+` Uses Python, which is considered easy to learn and is now frequently taught in both college and bootcamp programs.
+
+#### Cons
+
+- `-` Uses Python, which means learning a new programming language for those who haven't been exposed to it yet.
```

Comment:
I actually view Python as a pro, as it's one of the most popular programming languages out there (behind JS) and relatively quick to pick up.

Reply:
That's a good point, I went ahead and removed the line.